lib/open_project/scm/manageable_repository.rb

```diff
@@ -30,32 +30,47 @@
 module OpenProject
   module Scm
     module ManageableRepository
-      ##
-      # We let SCM vendor implementation define their own
-      # types (e.g., for differences in the management of
-      # local vs. remote repositories).
-      #
-      # But if they are manageable by OpenProject, they must
-      # expose this type through +available_types+.
-      MANAGED_TYPE = :managed
-      ##
-      # Reads from configuration whether new repositories of this kind
-      # may be managed from OpenProject.
-      def manageable?
-        !managed_root.nil?
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        ##
+        # We let SCM vendor implementation define their own
+        # types (e.g., for differences in the management of
+        # local vs. remote repositories).
+        #
+        # But if they are manageable by OpenProject, they must
+        # expose this type through +available_types+.
+        def managed_type
+          :managed
+        end
+
+        ##
+        # Reads from configuration whether new repositories of this kind
+        # may be managed from OpenProject.
+        def manageable?
+          ! (disabled_types.include?(managed_type) || managed_root.nil?)
```

Comment:
The first part of this condition duplicates the removal of disabled types that is done anyway.

IMO this should just read `!managed_root.nil?` or `!!managed_root`, whatever is your favorite cup of tea ^^